cpp/src/parquet/thrift_internal.h

```diff
@@ -411,17 +411,21 @@ class ThriftDeserializer {
       // thrift message is encrypted
       uint32_t clen;
       clen = *len;
+      if (clen > static_cast<uint32_t>(std::numeric_limits<int32_t>::max())) {
+        std::stringstream ss;
+        ss << "Cannot decrypt buffer with length " << clen << ", which overflows int32\n";
+        throw ParquetException(ss.str());
+      }
       // decrypt
-      auto decrypted_buffer = std::static_pointer_cast<ResizableBuffer>(
-          AllocateBuffer(decryptor->pool(),
-                         static_cast<int64_t>(clen - decryptor->CiphertextSizeDelta())));
-      const uint8_t* cipher_buf = buf;
+      auto decrypted_buffer = std::static_pointer_cast<ResizableBuffer>(AllocateBuffer(
```

Comment:
Unrelated, but since we're not attempting to resize the buffer, the cast to `ResizableBuffer` should be superfluous?

Reply:
`AllocateBuffer` here is `::parquet::AllocateBuffer` which returns a `std::shared_ptr<ResizableBuffer>`, rather than `::arrow::AllocateBuffer`, so it looks like the cast would be unnecessary even if we were resizing it:
https://github.com/apache/arrow/blob/031497ddab5ce497c49b87d7125bf5c01bf80f48/cpp/src/parquet/platform.cc#L36-L37

Reply:
Hmm, I see. We can try removing it at some point then, as the code currently looks weird.

Reply:
I've removed this as well as another occurrence of the same behaviour in https://github.com/apache/arrow/pull/43195/commits/f4a7721d13317c013b2b17aa099bb060c9fbe046